src/main/java/ubc/pavlab/rdp/services/GOServiceImpl.java

```diff
@@ -357,17 +381,59 @@ public GeneOntologyTermInfo getTerm( String goId ) {
         return getAncestorsInternal( term );
     }
 
-    private ConcurrentMap<GeneOntologyTermInfo, Set<GeneOntologyTermInfo>> ancestorsCache = new ConcurrentHashMap<>();
-
     private Collection<GeneOntologyTermInfo> getAncestorsInternal( GeneOntologyTermInfo term ) {
-        if ( ancestorsCache.containsKey( term ) )
-            return ancestorsCache.get( term );
-        Set<GeneOntologyTermInfo> results = new HashSet<>();
+        Cache ancestorsCache = cacheManager.getCache( ANCESTORS_CACHE_NAME );
+        //noinspection unchecked
+        Set<GeneOntologyTermInfo> results = ancestorsCache.get( term, Set.class );
+        if ( results != null )
+            return results;
+        results = new HashSet<>();
         for ( GeneOntologyTermInfo parent : getParents( term ) ) {
             results.add( parent );
             results.addAll( getAncestorsInternal( parent ) );
         }
         ancestorsCache.put( term, results );
         return results;
     }
+
+    private void evict( GeneOntologyTermInfo term ) {
+        evict( Collections.singleton( term ) );
+    }
+
+    private void evict( Iterable<GeneOntologyTermInfo> terms ) {
```

Comment:
It would be a good thing to consider at which `term.size()` it would be simply more efficient to use `evictAll()`.